File: packages/types/src/evaluation.ts
import { z } from "zod"

export const HireVerdict = z.enum(["hire", "lean-hire", "lean-no-hire", "no-hire"])
export type HireVerdict = z.infer<typeof HireVerdict>

export const DimensionScoreSchema = z.object({
  score: z.number().min(1).max(5),
  evidence: z.string(),
})

export const EvaluationSchema = z.object({
  id: z.string().uuid(),
  interviewId: z.string().uuid(),
  overallScore: z.number().min(1).max(5),
  dimensions: z.record(DimensionScoreSchema),
  hireVerdict: HireVerdict,
  strengths: z.array(z.string()),
  improvements: z.array(z.string()),
  deltaVsLast: z.number().nullable(),
  status: z.enum(["pending", "generating", "complete", "failed"]),
  createdAt: z.string().datetime(),
})

export type Evaluation = z.infer<typeof EvaluationSchema>

File: apps/api/models/evaluation.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import DateTime, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import JSONB, UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from core.database import Base


class Evaluation(Base):
    __tablename__ = "evaluations"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    interview_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("interviews.id"), nullable=False, unique=True
    )
    overall_score: Mapped[float] = mapped_column(nullable=True)
    dimension_scores: Mapped[dict] = mapped_column(JSONB, default=dict)
    hire_verdict: Mapped[str] = mapped_column(String(20), nullable=True)
    strengths: Mapped[dict] = mapped_column(JSONB, default=list)
    improvements: Mapped[dict] = mapped_column(JSONB, default=list)
    delta_vs_last: Mapped[float] = mapped_column(nullable=True)
    raw_evaluation: Mapped[str] = mapped_column(Text, nullable=True)
    model_used: Mapped[str] = mapped_column(String(50), nullable=True)
    status: Mapped[str] = mapped_column(String(20), default="pending")
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

    interview = relationship("Interview", back_populates="evaluation")
